huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account_privilege.go

```diff
@@ -0,0 +1,479 @@
+	var (
+		createGaussDBAccountPrivilegeHttpUrl = "v3/{project_id}/instances/{instance_id}/db-users/privilege"
+		createGaussDBAccountPrivilegeProduct = "gaussdb"
+	)
+	createGaussDBAccountPrivilegeClient, err := cfg.NewServiceClient(createGaussDBAccountPrivilegeProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating GaussDB Client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	accountName := d.Get("account_name").(string)
+	host := "%"
+	if h := d.Get("host").(string); h != "" {
+		host = h
+	}
+	createGaussDBAccountPrivilegePath := createGaussDBAccountPrivilegeClient.Endpoint + createGaussDBAccountPrivilegeHttpUrl
+	createGaussDBAccountPrivilegePath = strings.ReplaceAll(createGaussDBAccountPrivilegePath, "{project_id}",
+		createGaussDBAccountPrivilegeClient.ProjectID)
+	createGaussDBAccountPrivilegePath = strings.ReplaceAll(createGaussDBAccountPrivilegePath, "{instance_id}", instanceID)
+
+	createGaussDBAccountPrivilegeOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			201,
+		},
+	}
+	createGaussDBAccountPrivilegeOpt.JSONBody = utils.RemoveNil(buildCreateGaussDBAccountPrivilegeBodyParams(d, accountName, host))
+
+	var createGaussDBAccountPrivilegeResp *http.Response
+	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		createGaussDBAccountPrivilegeResp, err = createGaussDBAccountPrivilegeClient.Request("POST",
+			createGaussDBAccountPrivilegePath, &createGaussDBAccountPrivilegeOpt)
+		isRetry, err := handleGaussDBMysqlOperationError(err)
+		if isRetry {
+			return resource.RetryableError(err)
+		}
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return diag.Errorf("error creating GaussDB account privilege: %s", err)
+	}
+
+	createGaussDBAccountPrivilegeRespBody, err := utils.FlattenResponse(createGaussDBAccountPrivilegeResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	jobId, err := jmespath.Search("job_id", createGaussDBAccountPrivilegeRespBody)
+	if err != nil {
+		return diag.Errorf("error creating GaussDB MySQL account privilege: job_id is not found in API response")
+	}
+	err = waitForJobComplete(ctx, createGaussDBAccountPrivilegeClient, d.Timeout(schema.TimeoutCreate), instanceID, jobId.(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(instanceID + "/" + accountName + "/" + host)
+
+	return resourceGaussDBAccountPrivilegeRead(ctx, d, meta)
+}
+
+func buildCreateGaussDBAccountPrivilegeBodyParams(d *schema.ResourceData, accountName, host string) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":      accountName,
+		"host":      host,
+		"databases": buildCreateGaussDBAccountPrivilegeDatabasesChildBody(d),
+	}
+	params := map[string]interface{}{
+		"users": []interface{}{bodyParams},
+	}
+	return params
+}
+
+func buildCreateGaussDBAccountPrivilegeDatabasesChildBody(d *schema.ResourceData) []map[string]interface{} {
```

Comment:
```suggestion
func buildCreateGaussDBAccountPrivilegeDatabasesChildBody(databases []interface{}) []map[string]interface{} {
```
Only one code in this function uses `d`